### ISSUE TITLE
Using with tramp

### ISSUE THREAD

**Issue by jpmorgancalled** [Andersbakken/rtags#949]
I'm running Emacs running locally and running rdm on a remote machine. I indexed the code for my project on that remote machine. When I run Emacs on that machine, it has no problems doing symbol lookups.

But I have problems when running Emacs on my local machine, and using tramp to view the source code file on the remote machine. Running rtags commands like rtags-find-symbol gives this message:

`RTags: No results`

I'm using tramp with scp, not ssh. rdm is running on the remote machine.

This is in my init file:
```
(add-to-list 'tramp-remote-path "/rtags/path/bin")
(setq password-cache-expiry nil)
(setq rtags-tramp-enabled t)
```

Any idea what the problem might be?

**Comment by Andersbakken** [comment]
I've actually never tried the tramp setup myself with rtags. Can you verify that it's indeed running rc on the other machine? (like with ps or something)

**Comment by jpmorgancalled** [comment]
I can verify that rdm is running on the remote machine, yes. rc is not running, though.

**Comment by Andersbakken** [comment]
I was able to reproduce an issue with this. I'll try to see what's going on.

**Comment by vibrys** [comment]
Anders,
I can look at the the problem in a day or two if You don't mind.

**Comment by Andersbakken** [comment]
Don't mind at all. The help is much appreciated.

Anders

**Comment by vibrys** [comment]
@jpmorgancalled, does the following file exist on remote machine:
/var/run/user/<UID>/rdm.socket

**Comment by jpmorgancalled** [comment]
Nope. I install rtags locally, in my home directory, so that file wouldn't be created in /var anyway. And I cannot find such a file anywhere in my home directory.

**Comment by vibrys** [comment]
who runs rdm daemon? It must be remote user on remote machine.

/var/run/ is used by programs when they run, not when they are installed.

For example, when I do:

```mat@remote.host$ rdm```           # mat does not have root privilledges.

then the following file is created automatically:

```/scp:mat@remote.host:/var/run/user/1001/rdm.socket```

```/scp:mat@remote.host$ ll -d /var/run/user/1001```       #. as You see, priviledges fall into mat user.
```drwx------ 3 mat mat 80 Jun  3 14:22 /var/run/user/1001/```

Anyway, the case might vary from linux to linux, and as You mention, because rdm is not installed into the system, then maybe rdm creates this file at (see man pages):

~/.rdm

if yes, please customize ```rtags-socket-file``` to /home/user.on.remote.machine/.rdm

If not, then do one more step beforehand: explicitly tell rdm to use ~/.rdm with ```--socket-file``` parameter.

**Comment by jpmorgancalled** [comment]
I put this in my Emacs file on my local computer:
(setq rtags-socket-file "/remote/user/home/dir/.rdm")

But it has not changed the behavior. The .rdm file does exist on the remote machine when rdm is running on it (and this file has size 0, as you would expect I guess). I still cannot look up tags when accessing my remote files via Tramp though.

**Comment by vibrys** [comment]
Then I believe You do not see the following in *Messages* buffer, when trying to browse remote code then?:

```Can't seem to connect to server```

What does remote rdm tell when You jump to some location using ```rtags-find-symbol-at-point```?

for example mine tells something like that:
```
rc --current-file=/home/mat/CXX-trainingCode/11/p10.cpp -z --verify-version=122 -f /home/mat/CXX-trainingCode/11/p10.cpp:10:4: --all-targets
rc --current-file=/home/mat/CXX-trainingCode/11/p10.cpp -z --verify-version=122 --current-project
```

What does it tell when You run emacs on remote site directly?

The following format is fine.
```(setq rtags-socket-file "/remote/user/home/dir/.rdm")```
